scipy/optimize/_optimize.py

```diff
@@ -926,10 +981,10 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
             ind = np.argsort(fsim)
             sim = np.take(sim, ind, 0)
             fsim = np.take(fsim, ind, 0)
-            if callback is not None:
-                callback(sim[0])
             if retall:
                 allvecs.append(sim[0])
+            if _call_callback(callback, sim[0], fsim[0]):
+                break
```

Comment:
I am not sure about the name `_call_callback`. The line `if _call_callback_` to me reads as yes we are calling the callback, or no it did not run. Without looking at the function itself it's not clear that it's checking for stop iteration. Trying to think about another suggestion like `_call_callback_stop_iter`